### PR TITLE
feat: #90 Add accordion component

### PR DESCRIPTION
**Pull request by Shustovsky**

### CODE REVIEW

File: mw-webapp/src/component/accordion/Accordion.module.scss [hunk 1]
```diff
@@ -0,0 +1,18 @@
+@import "src/styles/variables";
+
+button,
```
Comment (tritonJS826) [inline mw-webapp/src/component/accordion/Accordion.module.scss:3]: lets use classNames

File: mw-webapp/src/component/accordion/Accordion.module.scss [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+h3 {
+    all: unset;
+}
+
+.AccordionRoot {
```
Comment (tritonJS826) [inline mw-webapp/src/component/accordion/Accordion.module.scss:8]: why with Upper letter?

File: mw-webapp/src/component/accordion/Accordion.module.scss [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    border-radius: 6px;
```
Comment (tritonJS826) [inline mw-webapp/src/component/accordion/Accordion.module.scss:9]: lets put border radius to variables

File: mw-webapp/src/component/accordion/Accordion.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+  /**
+   * The mode of operation for the accordion.
+   * @type {accordionTypes}
+   */
+  type: accordionTypes;
+  /**
+   * Additional custom class name for the component
+   */
+  className: string;
```
Comment (tritonJS826) [inline mw-webapp/src/component/accordion/Accordion.tsx:38]: why not optional?

File: mw-webapp/src/component/accordion/Accordion.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  /**
+   * An array of React elements representing the items to be displayed within the accordion.
+   */
+  children: ReactElement<AccordionItem>[];
+
+  /**
+   * The mode of operation for the accordion.
+   * @type {accordionTypes}
+   */
+  type: accordionTypes;
```
Comment (tritonJS826) [inline mw-webapp/src/component/accordion/Accordion.tsx:34]: should we set default one?

File: mw-webapp/src/component/accordion/Accordion.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  /**
+   * Additional custom class name for the component
+   */
+  className: string;
+}
+
+/**
+ * This component renders a vertically stacked set of interactive headings that each reveal an associated section of content.
+ */
+export function Accordion(props: AccordionProps) {
```
Comment (tritonJS826) [inline mw-webapp/src/component/accordion/Accordion.tsx:44]: lets use const for components

File: mw-webapp/src/component/accordion/Accordion.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  return (
+    <RadixAccordionRoot
+      className={clsx(styles.AccordionRoot, props.className)}
+      type={props.type}
+    >
+      {Children.map(props.children, (child, index) => {
```
Comment (tritonJS826) [inline mw-webapp/src/component/accordion/Accordion.tsx:50]: shold we use easier approach for example with appropriate props

File: mw-webapp/src/component/accordion/AccordionContent.module.scss [hunk 8]
```diff
@@ -0,0 +1,38 @@
+@import "src/styles/variables";
+
+.AccordionContent {
+    overflow: hidden;
+    font-size: 15px;
```
Comment (tritonJS826) [inline mw-webapp/src/component/accordion/AccordionContent.module.scss:5]: why custom font size

File: mw-webapp/src/component/accordion/AccordionContent.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import {Content} from "@radix-ui/react-accordion";
+import styles from "src/component/accordion/AccordionContent.module.scss";
+
+/**
+ * Props for the AccordionContent component.
```
Comment (tritonJS826) [inline mw-webapp/src/component/accordion/AccordionContent.tsx:5]: {@link ...}

File: mw-webapp/src/component/accordion/AccordionItem.tsx [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+ * A component representing an individual item within the Accordion.
+ * It should be used as a child of the Accordion component.
+ */
+export function AccordionItem(props: AccordionItem) {
+  return (
+    <RadixAccordionItem
+      className={styles.AccordionItem}
+      value={props.itemKey!}
+    >
+      {Children.map(props.children, (child) => child)}
```
Comment (tritonJS826) [inline mw-webapp/src/component/accordion/AccordionItem.tsx:32]: check other approaches to render lists

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/accordion/Accordion.module.scss (hunk 1 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/Accordion.module.scss]: done

File: mw-webapp/src/component/accordion/Accordion.module.scss (hunk 2 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/Accordion.module.scss]: done

File: mw-webapp/src/component/accordion/Accordion.module.scss (hunk 3 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/Accordion.module.scss]: done

File: mw-webapp/src/component/accordion/Accordion.tsx (hunk 4 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/Accordion.tsx]: done

File: mw-webapp/src/component/accordion/Accordion.tsx (hunk 5 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/Accordion.tsx]: done

File: mw-webapp/src/component/accordion/Accordion.tsx (hunk 7 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/Accordion.tsx]: done

File: mw-webapp/src/component/accordion/AccordionContent.module.scss (hunk 8 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/AccordionContent.module.scss]: done

File: mw-webapp/src/component/accordion/AccordionContent.tsx (hunk 9 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/AccordionContent.tsx:5]: done

File: mw-webapp/src/component/accordion/AccordionItem.tsx (hunk 10 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/AccordionItem.tsx]: done

File: mw-webapp/src/component/accordion/AccordionContent.tsx [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
```
Comment (Howl404) [inline mw-webapp/src/component/accordion/AccordionContent.tsx:5]: Maybe?

```suggestion
 * Props for the {@link AccordionContent} component.
```

Comment (Shustovsky) [reply mw-webapp/src/component/accordion/AccordionContent.tsx:5]: thx) I fixed it

File: mw-webapp/src/component/accordion/Accordion.tsx [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+/**
+ * This component renders a vertically stacked set of interactive headings that each reveal an associated section of content.
+ */
+export const Accordion = (props: AccordionProps) => {
+  return (
+    <RadixAccordionRoot
+      className={clsx(styles.accordionRoot, props.className)}
+      type={props.type || accordionTypes.single}
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/accordion/Accordion.tsx:52]: Why not ?? instead of ||?

File: mw-webapp/src/component/accordion/Accordion.tsx [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+    >
+      {props.items.map((item, index) => {
+        const itemKey = `item-${index}`;
+        return (
+          <AccordionItem
+            trigger={item.trigger}
+            content={item.content}
+            itemKey={itemKey}
+            key={itemKey}
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/accordion/Accordion.tsx:61]: Move out function tbefore return

File: mw-webapp/src/component/accordion/AccordionItem.tsx [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * A component representing an individual item within the Accordion.
+ * It should be used as a child of the Accordion component.
+ */
+export const AccordionItem = (props: AccordionItem) => {
+  return (
+    <RadixAccordionItem
+      className={styles.accordionItem}
+      value={props.itemKey}
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/accordion/AccordionItem.tsx:33]: Is it right that value = itemKey?

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/accordion/Accordion.tsx (hunk 12 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/Accordion.tsx:52]: ?? also good)
just || the first thing I thought about)

File: mw-webapp/src/component/accordion/AccordionItem.tsx (hunk 14 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/AccordionItem.tsx:33]: Yes, RadixAccordionItem must have unique value for the item.
Or would it be better to rename "ItemKey "?

File: mw-webapp/src/component/accordion/Accordion.tsx (hunk 13 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/Accordion.tsx:61]: done

File: mw-webapp/src/component/accordion/AccordionItem.tsx (hunk 14 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/accordion/AccordionItem.tsx:33]: No , no need rename

File: mw-webapp/src/component/accordion/Accordion.tsx (hunk 13 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/accordion/Accordion.tsx]: I was mean move out .map function

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/accordion/Accordion.tsx (hunk 13 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/Accordion.tsx]: I do it. but I'm not sure I got you right again)))

File: mw-webapp/src/component/accordion/Accordion.module.scss [hunk 15]
```diff
@@ -0,0 +1,8 @@
+@import "src/styles/variables";
+
+.accordionRoot {
+    border-radius: $accordion-radius;
+    margin: 1rem auto;
```
Comment (tritonJS826) [inline mw-webapp/src/component/accordion/Accordion.module.scss:5]: why 1 rem

File: mw-webapp/src/component/accordion/Accordion.tsx [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+      const itemKey = `item-${index}`;
+      return (
+        <AccordionItem
+          trigger={item.trigger}
+          content={item.content}
+          itemKey={itemKey}
+          key={itemKey}
+        />
+      );
+    });
```
Comment (tritonJS826) [inline mw-webapp/src/component/accordion/Accordion.tsx:60]: 1. index could not be key use new hook in (useId)
2. lets try to create pure function renderAccordionItem and move it outside of component, Lets make it simplier a little bit

File: mw-webapp/src/component/accordion/Accordion.tsx [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+   */
+  items: Array<{
+    trigger: ReactElement<AccordionTriggerProps>;
+    content: ReactElement<AccordionContentProps>;
+  }>;
+  /**
+   * The mode of operation for the accordion.
+   * @type {accordionTypes}
+   */
+  type?: accordionTypes;
```
Comment (tritonJS826) [inline mw-webapp/src/component/accordion/Accordion.tsx:38]: describe default

**Review by tritonJS826 — CHANGES_REQUESTED** [review]
Good. One more thing. Currently, you have 4 items in the same folder. Think about encapsulation, use directories to structure all this stuff

File: mw-webapp/src/component/accordion/Accordion.module.scss (hunk 15 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/Accordion.module.scss:5]: for beauty)
removed it)

File: mw-webapp/src/component/accordion/Accordion.tsx (hunk 16 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/Accordion.tsx]: done

File: mw-webapp/src/component/accordion/Accordion.tsx (hunk 17 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/Accordion.tsx:47]: done

File: mw-webapp/src/component/accordion/Accordion.module.scss (hunk 15 above)
Comment (tritonJS826) [reply mw-webapp/src/component/accordion/Accordion.module.scss]: I mean why relation and not absolute value?

File: mw-webapp/src/component/accordion/Accordion.module.scss [hunk 18]
```diff
@@ -0,0 +1,8 @@
+@import "src/styles/variables";
+
+.accordionRoot {
+    border-radius: $accordion-radius;
+    margin: 0 auto;
```
Comment (tritonJS826) [inline mw-webapp/src/component/accordion/Accordion.module.scss:5]: why does accordeon has auto margin. What if I don't want to center it?

**Review by tritonJS826 — APPROVED** [review]
Perfect! Just one more thing)

File: mw-webapp/src/component/accordion/AccordionContent/AccordionContent.module.scss [hunk 19]
```diff
@@ -1,4 +1,4 @@
-@import "src/styles/variables";
+@import "../../../styles/variables";
```
Comment (Howl404) [inline mw-webapp/src/component/accordion/AccordionContent/AccordionContent.module.scss:1]: Use absolute path

File: mw-webapp/src/component/accordion/AccordionTrigger/AccordionTrigger.module.scss [hunk 20]
```diff
@@ -1,4 +1,4 @@
-@import "src/styles/variables";
+@import "../../../styles/variables";
```
Comment (Howl404) [inline mw-webapp/src/component/accordion/AccordionTrigger/AccordionTrigger.module.scss:1]: Absolute path

**Review by Howl404 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/accordion/AccordionTrigger/AccordionTrigger.module.scss (hunk 20 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/AccordionTrigger/AccordionTrigger.module.scss:1]: Oh yes of course

File: mw-webapp/src/component/accordion/AccordionContent/AccordionContent.module.scss (hunk 19 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/AccordionContent/AccordionContent.module.scss:1]: Oh yes of course

File: mw-webapp/src/component/accordion/Accordion.module.scss (hunk 18 above)
Comment (Shustovsky) [reply mw-webapp/src/component/accordion/Accordion.module.scss:5]: Removed it

**Review by Howl404 — APPROVED** [review]

**Review by Ekaterina1994 — APPROVED** [review]

**Pull request merged**